src/Checkers/SourceCodeChecker.php

```diff
@@ -154,30 +153,32 @@ private function getMethodParameters(ReflectionMethod $method): array
     }
 
     /**
-     * Get the constant names and their values contained in the given class.
+     * Get the constant names and their values contained in the given reflection.
+     * This also includes cases from enums and their values (for string backed enums).
      *
-     * @param  ReflectionClass<object>  $class
+     * @param  ReflectionClass<object>  $reflection
      * @return array<int, string>
      */
-    private function getConstantNames(ReflectionClass $class): array
+    private function getConstantNames(ReflectionClass $reflection): array
     {
-        $constants = $class->getConstants();
+        $constants = $reflection->getConstants();
 
         return array_values(array_filter([
             ...array_keys($constants),
-            ...array_values($constants),
+            ...array_map(fn (mixed $value): mixed => $value instanceof BackedEnum && is_string($value->value) ? $value->value : $value, array_values($constants)),
         ], fn (mixed $values): bool => is_string($values)));
```

Comment:
I would perhaps look to clean this up or split it up in some way to make it more clear hat is going on. Perhaps a method for the initial purpose of the code, and perhaps a match statement within that method to show different functionality being used for the BackedEnum

Reply:
We could change it to this for example. Although I am unsure if this improves the readability that mutch?
Feedback or other suggestions are welcome!

```php
return array_values(array_filter([
    ...array_keys($constants),
    ...array_map(fn (mixed $value): mixed => match (true) {
        $value instanceof BackedEnum && is_string($value->value) => $value->value,
        default => $value,
    }, array_values($constants)),
], fn (mixed $values): bool => is_string($values)));
```